### PR TITLE
[feat][backend][2/x] sets up API routes for creating/joining/leaving team

### PR DESCRIPTION
**Pull request by WilhelmWillie**
see title, builds on top of #139 

testing these routes is a pain in the ass, ready for these to be reviewed but won't merge until serious further testing is done 

UPDATE: ended up building some interfaces to test API routes

Screen Shot 2019-11-26 at 4 58 26 PM
Screen Shot 2019-11-26 at 4 57 39 PM
Screen Shot 2019-11-26 at 4 58 59 PM

### CODE REVIEW

File: api/database/migrations/20191126000713-teams-add-dates.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+"use strict";
+
+module.exports = {
+  up: (queryInterface, Sequelize) => {
+    return Promise.all([
+      queryInterface.addColumn("Teams", "createdAt", {
```
Comment (WilhelmWillie) [inline api/database/migrations/20191126000713-teams-add-dates.js]: forgot to include this in the original migration oop

File: api/team.js [hunk 2]
```diff
@@ -1,13 +1,163 @@
...
+  await hackerProfile.setTeam(team);
+
+  return res.status(200).json({
+    message: "Successfully joined team"
+  });
+});
+
+// POST /api/team/leave
+// - If a hacker is on a team, attempt to leave that team
+router.post("/leave", async (req, res) => {
```
Comment (WilhelmWillie) [inline api/team.js]: i haven't tested this one yet, will do in the next PR once i build out the interface further

File: api/team.js [hunk 3]
```diff
@@ -1,13 +1,163 @@
...
+      }
+    ]
+  });
+
+  if (team) {
+    return res.json({ team });
+  } else {
+    return res.json({
+      message: "User does not currently belong to a team"
+    });
```
Comment (saresend) [inline api/team.js]: Probably handled on the frontend, but maybe having a nullable "team"  attribute that we can check might be a bit cleaner than having this return variably shaped JSON depending on if a team exists. Alternately, 404 just so its easier to check?

File: api/team.js [hunk 4]
```diff
@@ -1,13 +1,163 @@
...
+// GET /api/team/:code
+// - If provided a team code, retrieve it
+router.get("/:code", async (req, res) => {
+  // Try to find a team with the provided code
+  const team = await models.Team.findOne({
+    where: { teamCode: req.params.code || "" }
+  });
+  if (!team) {
+    return res
+      .status(400)
```
Comment (saresend) [inline api/team.js]: Nit: 404?

File: api/team.js [hunk 5]
```diff
@@ -1,13 +1,163 @@
...
+  if (team) {
+    return res.status(500).json({ message: "User already belongs on a team" });
+  }
+
+  let generatedCode = Math.random()
+    .toString(36)
+    .slice(4, 8)
+    .toUpperCase();
+
+  while (
```
Comment (saresend) [inline api/team.js]: yeehaw

**Pull request merged**